### ISSUE TITLE
[2.8] Stale secrets for impersonation user in downstream cluster

### ISSUE THREAD

**Issue by samjustus** [rancher/rancher#46537]
backport https://github.com/rancher/rancher/issues/44347

**Comment by dasarinaidu** [comment]
This issue is not fixed on `v2.8-68e5d43e449fcd3ef73baf1caee98da6f0104eed-head`

I still see stale secrets :

```
>  k get secrets -n cattle-impersonation-system | grep -i  user-h7hms
cattle-impersonation-user-h7hms-token-6r72c     kubernetes.io/service-account-token   3      3m57s
cattle-impersonation-user-h7hms-token-8kgwk     kubernetes.io/service-account-token   3      3m57s
cattle-impersonation-user-h7hms-token-tgq75     kubernetes.io/service-account-token   3      3m57s
>  k get secrets -n cattle-impersonation-system | grep -i  u-ocjra4tnyd
cattle-impersonation-u-ocjra4tnyd-token-gwf6f   kubernetes.io/service-account-token   3      4m18s
cattle-impersonation-u-ocjra4tnyd-token-n72zv   kubernetes.io/service-account-token   3      4m18s
>  k get secrets -n cattle-impersonation-system | grep -i u-b3ub6bhfl5 
cattle-impersonation-u-b3ub6bhfl5-token-7snzz   kubernetes.io/service-account-token   3      4m2s
cattle-impersonation-u-b3ub6bhfl5-token-f5kzj   kubernetes.io/service-account-token   3      4m2s
```

**Comment by dasarinaidu** [comment]
This issue is partially fixed and opened a new issue for the remaining issue. I m closing this ticket as there are no regression issues with this fix. https://github.com/rancher/rancher/issues/47126
